projects/dotcms-field-elements/src/components/dot-tags/dot-tags.tsx

```diff
@@ -0,0 +1,173 @@
+                            label={tagLab}
+                            disabled={this.disabled}
+                            onRemove={this.removeTag.bind(this)}
+                        >
+                        </dot-chip>
+                    )}
+                </div>
+
+                <dot-autocomplete
+                    id={this.name}
+                    class={getErrorClass(this.status.dotValid)}
+                    disabled={this.disabled || null}
+                    placeholder={this.placeholder}
+                    threshold={this.threshold}
+                    debounce={this.debounce}
+                    data={this.getData.bind(this)}
+                    onLostFocus={() => this.blurHandler()}
+                    onSelection={(event) => this.addTag(event.detail)}
+                >
+                </dot-autocomplete>
+
+                {getTagHint(this.hint)}
+                {getTagError(this.showErrorMessage(), this.getErrorMessage())}
+            </Fragment>
+        );
+    }
+
+    @Watch('value')
+    setValue(): void {
+        this.status = updateStatus(this.status, {
+            dotTouched: true,
+            dotPristine: false,
+            dotValid: this.isValid()
+        });
+
+        this.emitValueChange();
+        this.emitStatusChange();
+    }
+
+    private addTag(label: string): void {
+        const values = this.getValues();
+
+        if (!values.includes(label)) {
+            values.push(label);
+            this.value = values.join(',');
+            this.selected.emit(label);
+        }
+    }
+
+    private removeTag(event: CustomEvent): void {
+        const values = this.getValues().filter(item => item !== event.detail);
+        this.value = values.join(',');
+        this.removed.emit(event.detail);
+    }
+
+    private getValues(): string[] {
+        return this.value ? this.value.split(',') : [];
+    }
+
+    private isValid(): boolean {
+        return !this.required || (this.required && !!this.value);
+    }
+
+    private showErrorMessage(): boolean {
+        return this.getErrorMessage() && !this.status.dotPristine;
+    }
+
+    private getErrorMessage(): string {
+        return this.isValid()
+                ? ''
+                : this.requiredMessage;
+    }
+
+    private blurHandler(): void {
+        if (!this.status.dotTouched) {
+            this.status = updateStatus(this.status, {
+                dotTouched: true
+            });
+            this.emitStatusChange();
+        }
+    }
+
+    private emitStatusChange(): void {
+        this.statusChange.emit({
+            name: this.name,
+            status: this.status
+        });
+    }
+
+    private emitValueChange(): void {
+        this.valueChange.emit({
+            name: this.name,
+            value: this.value
+        });
+    }
+
+    private async getData(): Promise<String[]> {
+        const source = await fetch(
+            'https://tarekraafat.github.io/autoComplete.js/demo/db/generic.json'
```

Comment:
what is this?

Reply:
it is for get demo data for now meanwhile we wait for the back end